main/cmds.go

```diff
@@ -1,17 +1,17 @@
 package main
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
-	"strconv"
 
 	"github.com/Azure/azure-docker-extension/pkg/vmextension"
 	"github.com/Azure/custom-script-extension-linux/pkg/seqnum"
 	"github.com/go-kit/kit/log"
 	"github.com/pkg/errors"
 )
 
-type cmdFunc func(*log.Context, vmextension.HandlerEnvironment) error
+type cmdFunc func(ctx *log.Context, hEnv vmextension.HandlerEnvironment, seqNum int) error
```

Comment:
It's unfortunate you have to pass in another parameter.  You really want the union of hEnv and seqNum as you had before.  I would consider creating the type you want to hold all of the necessary state.